packages/plot/src/marks/GeoMark.js

```diff
@@ -22,12 +23,17 @@ export class GeoMark extends Mark {
 
     // parse GeoJSON strings to JSON objects
     const geom = this.channelField('geometry')?.as;
+
+    if (isArrowTable(data)) {
+      this.data = data.toArray();
```

Comment:
Needed since we are modifying the array. May need some more thinking to change this if we want to work with arrow.